### PR TITLE
test(docs): add visual testing using applitools

### PR DESCRIPTION
**Pull request by Nkemjiks**
**Description:**
Hi everyone, this PR adds visual regression testing using (Applitool)[https://applitools.com/] eyes. If you are unfamiliar with Visual Regression testing, here a list of references from Applitools to learn more about it.

https://applitools.com/docs/topics/overview/overview-visual-testing.html
https://applitools.com/blog/troubleshoot-fix-react-bugs
https://circleci.com/gh/myspivey/lock/6

As an open source tool, this service will be offered to you free with an OSS license. They do offer a generous free tier if you want to get a license now to play around with it. I used protractor as the browser test harness because Applitools has an SDK for it.

The process for this testing is the docs site is served using concurrently and the existing design script, then protractors launches a headless Chrome where it runs the tests. Applitool's SDK then takes DOM snapshots w/ styles (not screenshots) from that headless Chrome and sends them to Applitool's servers where it renders those results in Chrome and Firefox and then takes a screenshot of that result and does its intelligent comparisons against the baseline. The baseline is created the first time you run it. If there's an issue, it'll be reported in CircleCI with an included link directly to Applitool's web app where you can see the visual differences and either accept or reject them or do other cool things like set ignored regions or tweak the matching behavior.

You'll need to create an Applitools account and export `APPLITOOLS_API_KEY=Your_API_Key_Here` in your local environment if you run it locally.

To run the test locally, go to your directory and run:
`npm run e2e`

**Related issue (if exists):**

### CODE REVIEW

**Comment by ladyleet** [comment]
@benlesh do you have an idea of when this will get merged?

File: docs_app/tests/e2e/visual-testing.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { browser } from 'protractor';
+
+const Eyes = require('eyes.selenium').Eyes;
```
Comment (JWO719) [inline docs_app/tests/e2e/visual-testing.e2e-spec.ts]: It seems like this isn't added to the dependency list in the package.json.

**Comment by Nkemjiks** [comment]
Hi @JWO719 , thanks for pointing out the missing dependency. I was having issues running the already included e2e test and was doing alot of reverting back. I had installed the dependency locally before reverting but i didn't delete my node_modules that is why i didn't get the errors. I have updated the PR to include the dependency. Please pull and run again. Do let me know if you have any more issues

**Comment by ladyleet** [comment]
@JWO719 new status at all? // @Nkemjiks @myspivey

**Comment by JWO719** [comment]
Hi @Nkemjiks, still I'm not able to run the tests locally. I'm still running in the same error, described above. I know remote debugging is kinda hard. If you like to we can do a screensharing session and investigate together what's causing the error?

**Comment by Nkemjiks** [comment]
> Hi @Nkemjiks, still I'm not able to run the tests locally. I'm still running in the same error, described above. I know remote debugging is kinda hard. If you like to we can do a screensharing session and investigate together what's causing the error?

What time will you be available tomorrow so we can look into this?

**Comment by JWO719** [comment]
Sorry tomorrow might be a little bit difficult. It's propably easier to make an appointment more in private. You can contact me on twitter @niklas_wortmann or in the reactivex slack channel (I think @ladyleet could invite you)

**Comment by JWO719** [comment]
Together with @Nkemjiks we found out that my local issues were caused by an angular/cli [bug](https://github.com/angular/angular-cli/issues/13113 )
Unfortunately to fix this, we need to merge #4688 before. I will focus on merging #4688 so that I can test this one again. Sorry for keeping this on hold for the meantime.

**Comment by benlesh** [comment]
Ping @JWO719

**Comment by JWO719** [comment]
So the blocking pr is finally merged (sorry for the delay): I gave it a quick try and noticed some things:
 - this pr needs a rebase
 - due to some compilation errors the other e2e spec files needs to be deleted

@Nkemjiks Could you take care of it? Many thanks for your contribution!

**Comment by Nkemjiks** [comment]
@JWO719 , I have rebased and deleted the other spec file as request. I don't know why the typescript check is failing though..

**Comment by cartant** [comment]
@Nkemjiks I wouldn't worry about the TypeScript error. The release of TypeScript 3.6 has effected some errors with RxJS due to what appears to be a TypeScript bug - see #4992. (It looks like the CI build script runs `npm i` - which, IIRC, upgrades to the latest versions that match the semver range, so the CI build is most likely using TS 3.6.)

@JWO719 Let's ignore the TypeScript error and merge this if everything else is good to go.

**Comment by JWO719** [comment]
Thank you very much @Nkemjiks for all the work you put into that and I'm really sorry for all the hassle!

**Pull request merged**